dags/utils/inventory.py

```diff
@@ -0,0 +1,112 @@
+import boto3
+import json
+import gzip
+import csv
+import os.path as op
+from copy import deepcopy
+from datetime import datetime, timedelta
+from airflow.contrib.hooks.aws_hook import AwsHook
+
+from urllib.parse import urlparse
+
+
+class s3:
+    def __init__(self, url, conn, region, suffix=''):
+        self.url = url
+        self.suffix = suffix
+        aws_hook = AwsHook(aws_conn_id=conn)
+        cred = aws_hook.get_session().get_credentials()
+        self.s3 = boto3.client('s3', aws_access_key_id=cred.access_key ,
+                                aws_secret_access_key=cred.secret_key  ,
+                                region_name=region)
+        self.bucket = self.urlparse(self.url)['bucket']
+
+    # Modified derived from https://alexwlchan.net/2018/01/listing-s3-keys-redux/
```

Comment:
- Check out an updated version of reference resource with efficient code and using paginators - [https://alexwlchan.net/2019/07/listing-s3-keys](https://alexwlchan.net/2019/07/listing-s3-keys)
- Have you look into `s3-find` ODC tool that lists S3 bucket with wildcard - [https://github.com/opendatacube/odc-tools](https://github.com/opendatacube/odc-tools)

Reply:
I don't want to list buckets as it's extremely expensive in S2 case.  I'm simply parsing inventory files which is extremely fast compared to listing.